sdk/metric/internal/aggregate/aggregate.go

```diff
@@ -42,6 +42,17 @@ type Builder[N int64 | float64] struct {
 	Filter attribute.Filter
 }
 
+func (b Builder[N]) filter(f Measure[N]) Measure[N] {
+	if b.Filter != nil {
+		fltr := b.Filter // Copy to prevent any funny stuff.
```

Comment:
maybe something like

```suggestion
		fltr := b.Filter // Copy to make it immutable after assignment.
```